pennylane_orquestra/orquestra_device.py

```diff
@@ -0,0 +1,378 @@
+            supports_inverse_operations=False,
+            supports_analytic_computation=True,
+            returns_probs=False,
+        )
+        return capabilities
+
+    @property
+    def backend_specs(self):
+        """The backend specifications defined for the device.
+
+        Returns:
+            str: the backend specifications represented as a string
+        """
+        if self._backend_specs is None:
+            backend_specs_dict = self.create_backend_specs()
+            self._backend_specs = json.dumps(backend_specs_dict)
+
+        return self._backend_specs
+
+    def create_backend_specs(self):
+        """Create the backend specifications based on the device options.
+
+        Returns:
+            dict: the backend specifications represented as a dictionary
+        """
+        backend_specs = {}
+        backend_specs["module_name"] = self.qe_module_name
+        backend_specs["function_name"] = self.qe_function_name
+
+        if self.backend is not None:
+            # Only backends that have multiple backend need to specify one
+            # E.g., qe-qiskit
+            backend_specs["device_name"] = self.backend
+
+        if not self.analytic:
+            backend_specs["n_samples"] = self.shots
+
+        return backend_specs
+
+    @property
+    def latest_id(self):
+        """Returns the latest workflow ID that has been executed.
+
+        Returns:
+            str: the ID of the latest workflow that has been submitted
+        """
+        return self._latest_id
+
+    @property
+    def filenames(self):
+        """Returns the names of the workflow files created during device
+        executions.
+
+        Returns:
+            list: the workflow filenames
+        """
+        return self._filenames
+
+    @property
+    @abc.abstractmethod
+    def qe_module_name(self):
+        """Device specific Orquestra module name used in the backend
+        specification."""
+
+    @property
+    @abc.abstractmethod
+    def qe_function_name(self):
+        """Device specific Orquestra function name used in the backend
+        specification."""
+
+    @property
+    @abc.abstractmethod
+    def qe_component(self):
+        """Device specific Orquestra component name used in the backend
+        specification."""
+
+    def serialize_circuit(self, circuit):
+        """Serializes the circuit before submission according to the backend
+        specified.
+
+        The circuit is represented as an OpenQASM 2.0 program. Measurement
+        instructions are removed from the program as the operator is passed
+        separately.
+
+        Args:
+            circuit (~.CircuitGraph): circuit to serialize
+
+        Returns:
+            str: OpenQASM 2.0 representation of the circuit without any
+            measurement instructions
+        """
+        qasm_str = circuit.to_openqasm(rotations=not self.analytic)
+
+        qasm_without_measurements = re.sub("measure.*?;\n", "", qasm_str)
```

Comment:
Nice, nice! :fire: 

Just a thought on this: could there ever be statements after the measure statement on the same line? E.g. `measure bla bla; gate bla;`. In that case I think this would work instead (greedily checking for an extra `\n` and also removing any whitespace before the following statement on the same line):

```suggestion
        qasm_without_measurements = re.sub("measure.*?;\n?\s*", "", qasm_str)
```

Reply:
Think at the moment no, as PennyLane doesn't allow applying gates after the measurements, but could come in handy. Thanks!